### PR TITLE
Updating section and field product fills to support multiple slots

### LINKED ISSUE #36609: [Product Extensibility]: Add support for each product fill to fill multiple slots
Right now the `WooProductFieldItem` and `WooSectionFieldItem` component accept only a single ID of a location (field -> section, and section -> tab) to fill. We'll need to support multiple.

**Acceptance Criteria**
- Change the property names to make them plural (ie section -> sections)
- These properties should now accept an array of objects, with each object including a `name` and `order`

### PR DESCRIPTION
**Pull request by joelclimbsthings**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adding support to all product editor slot fills for multiple targets.

Closes #36609  .

### How to test the changes in this Pull Request:

1. Checkout branch.
2. Enable the `product-variation-management` feature flag using the WC Beta Tester Plugin
3. Enable the new product management experience using WooCommerce -> Settings -> Advanced -> Features -> Try the new product editor.
4. Go to Products -> Add New
5. Smoke test all tabs and fields.
6. Create an option on the Options tab, and then click edit
7. Ensure the tabs on the variant are working as expected

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/components/src/woo-product-field-item/woo-product-field-item.tsx [hunk 1]
```diff
@@ -24,28 +24,35 @@ type WooProductFieldSlotProps = {
...
+					{ ( fillProps: Fill.Props ) => {
+						return createOrderedChildren<
+							Fill.Props & SlotContextHelpersType,
+							{ _id: string }
+						>(
+							children,
+							sectionOrder || 20,
+							{
+								...fillProps,
+								...getFillHelpers(),
```
Comment (louwie17) [inline packages/js/components/src/woo-product-field-item/woo-product-field-item.tsx:52]: I think we should expose the `sectionName` here as well.

File: packages/js/components/src/woo-product-section-item/woo-product-section-item.tsx [hunk 2]
```diff
@@ -3,41 +3,50 @@
...
+			{ tabs.map( ( { name: tabName, order: tabOrder } ) => (
+				<Fill
+					name={ `woocommerce_product_section_${ tabName }` }
+					key={ tabName }
+				>
+					{ ( fillProps: Fill.Props ) => {
+						return createOrderedChildren< Fill.Props >(
+							children,
+							tabOrder || 20,
+							fillProps
```
Comment (louwie17) [inline packages/js/components/src/woo-product-section-item/woo-product-section-item.tsx]: Exposing the `tabName` as part of the `fillProps` would be good to.

File: packages/js/data/src/product-form/resolvers.ts [hunk 3]
```diff
@@ -34,10 +29,6 @@ export function* getFields() {
 	}
 }
 
-export function* getCountry() {
-	yield resolveSelect( STORE_NAME, 'getProductForm' );
-}
```
Comment (louwie17) [inline packages/js/data/src/product-form/resolvers.ts:39]: Are these not being used anymore? or was this added by accident?

Comment (joelclimbsthings) [reply packages/js/data/src/product-form/resolvers.ts:39]: It looks like it was just by accident that I included this when I originally created the data store.

File: packages/js/components/src/woo-product-field-item/woo-product-field-item.tsx (hunk 1 above)
Comment (joelclimbsthings) [reply packages/js/components/src/woo-product-field-item/woo-product-field-item.tsx:52]: Done in e6543756c253907b96119baedec6401cd0448076

File: packages/js/components/src/woo-product-section-item/woo-product-section-item.tsx (hunk 2 above)
Comment (joelclimbsthings) [reply packages/js/components/src/woo-product-section-item/woo-product-section-item.tsx]: Done in e6543756c253907b96119baedec6401cd0448076

**Comment by joelclimbsthings** [comment]
Thanks @louwie17 , I've refactored this to only include the relevant changes and your requested revisions. I'll do the changes specific to variants in another PR.

**Review by louwie17 — APPROVED** [review]
This tested well, and thanks for updating this @joelclimbsthings, nice work!!

**Pull request merged**